analytics.go

```diff
@@ -100,6 +108,10 @@ func NewWithConfig(writeKey string, dataPlaneUrl string, config Config) (cli Cli
 	return
 }
 
+func (c *client) SetGZIP(gzip bool) {
```

Comment:
`Set` `Get` methods are not very popular/idiomatic in Go, maybe:

```suggestion
func (c *client) WithGZIP(gzip bool) {
```